tests/virtualization/xen/virtmanager_add_devices.pm

```diff
@@ -76,7 +78,7 @@ sub run {
 
             assert_and_click 'virt-manager_graphical-console';
 
-            detect_login_screen();
+            detect_login_screen() if (!check_screen('virt-manager_viewer_disconnected', 5));
```

Comment:
For my future self: I think this is something we should investigate why this is happening.